### PR TITLE
feat: Flutter engine integration

### PR DESCRIPTION
**Pull request by LuckZAE**
## Summary

Adds a FlutterChannel communication layer between C++ and the Flutter engine, replacing direct FlutterViewHost::MethodCallHandler usage with a structured message-passing pipeline.

### What's Added

- **FlutterChannel** — bidirectional method call channel with:
  - Namespace routing (`ns("prefix").on("method")`) for organized handler registration
  - Route table mapping `"prefix.method"` → handler functions
  - Message queue: buffers up to 256 outbound messages until Dart signals `system.ready`

- **Middleware pipeline** — three built-in interceptors:
  - `LoggingMiddleware` — assigns a monotonic request ID per inbound call
  - `ThreadGuardMiddleware` — ensures handler execution stays on the wxWidgets UI thread, deferring to `CallAfter` when needed
  - `TimeoutMiddleware` — fires a TIMEOUT reply via `wxTimer` if a handler doesn't respond within N seconds

- **Tests** (`test_flutter.cpp`) — covering channel lifecycle, namespace routing, middleware chain order, invoke-before-ready queuing, and edge cases

### Files Changed

| New | Modified | Removed |
|-----|----------|---------|
| `src/slic3r/GUI/Flutter/FlutterChannel.{h,cpp}` | `src/CMakeLists.txt` | `tests/libslic3r/test_flutter_platform.cpp` |
| `src/slic3r/GUI/Flutter/FlutterMiddleware.{h,cpp}` | `tests/libslic3r/CMakeLists.txt` | |
| `tests/libslic3r/test_flutter.cpp` | `src/slic3r/GUI/Flutter/FlutterPanel.hpp` | |

### Design

```
Dart MethodChannel
    ↕ invoke / Reply (StandardMethodCodec)
FlutterViewHost
    ↕
Middleware chain (Log → ThreadGuard → Timeout)
    ↕
Route table [prefix.method → Handler]
```

**Inbound** (Flutter → C++): message enters the middleware chain, each middleware can intercept, defer to UI thread, or set a timeout. If all pass, the registered route handler is invoked.

**Outbound** (C++ → Flutter): `channel.invoke(method, args)` sends through the view. Messages sent before Dart signals `system.ready` are queued and flushed automatically.

### CODE REVIEW

File: src/slic3r/GUI/Flutter/FlutterMiddleware.cpp [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+                          public wxEvtHandler {
+    int m_sec;
+public:
+    explicit TimeoutMiddleware(int s) : m_sec(s) {}
+
+    bool onInbound(const std::string&, const std::string&,
+                   FlutterViewHost::Reply reply) override {
+        auto fn    = std::make_shared<FlutterViewHost::Reply>(std::move(reply));
+        auto fired = std::make_shared<bool>(false);
+        auto t     = std::make_unique<wxTimer>(this);
```
Comment (SukiSunYuhang) [inline src/slic3r/GUI/Flutter/FlutterMiddleware.cpp:62]: handler 正常返回后主动取消对应的 timer 即可，这样写 m_timers 会无限增长

File: src/slic3r/GUI/Flutter/FlutterChannel.h [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    std::unordered_map<std::string, Handler> m_routes;
+    std::vector<std::shared_ptr<Middleware>> m_middleware;
+    FlutterViewHost* m_view = nullptr; // main-thread only
+    bool m_handlerCalled = false;
+};
+
+// Middleware factory functions
+std::shared_ptr<FlutterChannel::Middleware> makeLoggingMiddleware();
+std::shared_ptr<FlutterChannel::Middleware> makeThreadGuardMiddleware();
+std::shared_ptr<FlutterChannel::Middleware> makeTimeoutMiddleware(int seconds);
```
Comment (SukiSunYuhang) [inline src/slic3r/GUI/Flutter/FlutterChannel.h]: 建议用 makeTimeoutMiddleware (std::chrono::seconds seconds)

File: src/slic3r/GUI/Flutter/FlutterChannel.h [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+#include <cassert>
+
+class FlutterChannel {
+public:
+    using Handler = std::function<void(const std::string&, FlutterViewHost::Reply)>;
+    using Reply   = FlutterViewHost::Reply;
+    using MiddlewarePtr = std::shared_ptr<class Middleware>;
+
+    explicit FlutterChannel(const std::string& name);
+    FlutterChannel(const FlutterChannel&) = delete;
```
Comment (SukiSunYuhang) [inline src/slic3r/GUI/Flutter/FlutterChannel.h:19]: 显式删除了拷贝构造/赋值，但没有声明移动构造/赋值，建议显式加上移动构造
